kernel/src/util/net/options/utils.rs

```diff
@@ -138,8 +139,11 @@ impl WriteToUser for LingerOption {
     }
 }
 
+const TCP_CONGESTION_NAME_MAX: u32 = 16;
+
 impl ReadFromUser for CongestionControl {
-    fn read_from_user(addr: Vaddr, max_len: u32) -> Result<Self> {
+    fn read_from_user(addr: Vaddr, mut max_len: u32) -> Result<Self> {
+        max_len = (TCP_CONGESTION_NAME_MAX - 1).min(max_len);
         let mut bytes = vec![0; max_len as usize];
         current_userspace!().read_bytes(addr, &mut VmWriter::from(bytes.as_mut_slice()))?;
         let name = String::from_utf8(bytes).unwrap();
```

Comment:
This `unwrap` is reachable if the user program does not pass a valid UTF-8 string. This is not a problem introduced by this PR, though.